src/multistore.rs

```diff
@@ -74,35 +77,36 @@ impl<E: Element> Store<E> for MultiStore<E> {
 
     fn read_range(&self, r: Range<usize>) -> Result<Vec<E>> {
         if r.start > DISK_MAX {
+            // entire range is in mem
             let nr = Range {
                 start: r.start - DISK_MAX,
                 end: r.end - DISK_MAX,
             };
-            self.1.read_range(nr)
+            self.mem.read_range(nr)
+        } else if r.end > DISK_MAX {
+            // split across disk and mem
+            let nrdisk = Range {
+                start: r.start,
+                end: DISK_MAX,
+            };
+            let nrmem = Range {
+                start: 0,
+                end: r.end - DISK_MAX,
+            };
+            let rdisk = self.mem.read_range(nrdisk).unwrap();
```

Comment:
I think I've got it now ... I was getting the same result commp regardless of whether this was switched around. My initial version of this had a panic when you queried a split range because <too hard> but when I ran it, it panicked, so I had to implement it! So those two facts combined had me confused and worried because it seemed like it didn't matter what data was being fed.

However, I've gone over the code again and I _think_ I had `>` and `>=` wrongly used, I also implemented the split copy in `copy_from_slice` which is essentially the write version of this `range_read` (see 6ae7072). But now I have logging in both of those split blocks and it's no longer logging. I can't recall where I originally got the size of the `DISK_MAX` from, `262144 * 48` but I think 262144 was from inspecting how it worked originally and it seeming to work on 262144 chunks. Multiplied by 48 gave me roughly how much disk I wanted to use up. So 🤞 I think I'm hitting right on aligned slices and not needing these split read/write operations anymore. Hopefully my code now would survive if it did, though, but it's another reason we need to do a lot of testing of this stuff.